pkg/metamorph/client.go

```diff
@@ -196,23 +196,38 @@ func (m *Metamorph) SubmitTransaction(ctx context.Context, tx *sdkTx.Transaction
 		}, nil
 	}
 
-	response, err := m.client.PutTransaction(ctx, request)
-	if err != nil {
-		m.logger.Warn("Failed to submit transaction", slog.String("hash", tx.TxID()), slog.String("key", err.Error()))
-		if m.mqClient != nil {
-			err := m.mqClient.PublishMarshal(SubmitTxTopic, request)
-			if err != nil {
-				return nil, err
+	var response *metamorph_api.TransactionStatus
+	var err error
+	// in case of error try PutTransaction until timeout expires
+	for {
+		start := time.Now()
```

Comment:
Doesn't this need to be defined outside of the for loop?

Also it should use the now function of the object `start := m.now()`

Reply:
No, we are measuring `PutTransaction` call duration on every iteration